src/hl/dataset.rs

```diff
@@ -140,728 +168,1112 @@ impl Dataset {
-                ($ds:expr, $tp:ty, $v:expr) => {
-                    assert_eq!(($ds).fill_value::<$tp>().unwrap(), Some(($v) as $tp));
-                };
-            }
+    ////////////////////
+    // LinkCreate     //
+    ////////////////////
 
-            macro_rules! check_fill_value_approx {
-                ($ds:expr, $tp:ty, $v:expr) => {{
-                    let fill_value = ($ds).fill_value::<$tp>().unwrap().unwrap();
-                    // FIXME: should inexact float->float casts be prohibited?
-                    assert!((fill_value - (($v) as $tp)).abs() < (1.0e-6 as $tp));
-                }};
-            }
+    pub fn set_link_create_plist(&mut self, lcpl: &LinkCreate) {
+        self.lcpl_base = Some(lcpl.clone());
+    }
 
-            macro_rules! check_all_fill_values {
-                ($ds:expr, $v:expr) => {
-                    check_fill_value!($ds, u8, $v);
-                    check_fill_value!($ds, u16, $v);
-                    check_fill_value!($ds, u32, $v);
-                    check_fill_value!($ds, u64, $v);
-                    check_fill_value!($ds, i8, $v);
-                    check_fill_value!($ds, i16, $v);
-                    check_fill_value!($ds, i32, $v);
-                    check_fill_value!($ds, i64, $v);
-                    check_fill_value!($ds, usize, $v);
-                    check_fill_value!($ds, isize, $v);
-                    check_fill_value_approx!($ds, f32, $v);
-                    check_fill_value_approx!($ds, f64, $v);
-                };
-            }
+    pub fn set_lcpl(&mut self, lcpl: &LinkCreate) {
+        self.set_link_create_plist(lcpl);
+    }
+
+    pub fn link_create_plist(&mut self) -> &mut LinkCreateBuilder {
+        &mut self.lcpl_builder
+    }
 
-            let ds = file.new_dataset::<u16>().create_anon(100).unwrap();
-            check_all_fill_values!(ds, 0);
+    pub fn lcpl(&mut self) -> &mut LinkCreateBuilder {
+        self.link_create_plist()
+    }
 
-            let ds = file.new_dataset::<u16>().fill_value(42).create_anon(100).unwrap();
-            check_all_fill_values!(ds, 42);
+    pub fn with_link_create_plist<F>(&mut self, func: F)
+    where
+        F: Fn(&mut LinkCreateBuilder) -> &mut LinkCreateBuilder,
+    {
+        func(&mut self.lcpl_builder);
+    }
 
-            let ds = file.new_dataset::<f32>().fill_value(1.234).create_anon(100).unwrap();
-            check_all_fill_values!(ds, 1.234);
-        })
+    pub fn with_lcpl<F>(&mut self, func: F)
+    where
+        F: Fn(&mut LinkCreateBuilder) -> &mut LinkCreateBuilder,
+    {
+        self.with_link_create_plist(func);
+    }
+
+    // LCPL properties
+
+    pub fn create_intermediate_group(&mut self, create: bool) {
+        self.with_lcpl(|pl| pl.create_intermediate_group(create));
     }
+
+    pub fn char_encoding(&mut self, encoding: CharEncoding) {
+        self.with_lcpl(|pl| pl.char_encoding(encoding));
+    }
+}
+
+macro_rules! impl_builder_stuff {
```

Comment:
Wonder - are the generated docs ok with these macros?

Reply:
The docs are replicated on all relevant types, this is duplication, but it all shows as a separate impl block. I've added a small comment on the `impl` blocks to describe how they are all common between dataset-builders